storage/local/interface.go

```diff
@@ -26,7 +26,8 @@ import (
 // Storage ingests and manages samples, along with various indexes. All methods
 // are goroutine-safe. Storage implements storage.SampleAppender.
 type Storage interface {
-	Querier
+	// Querier returns a new Querier on the storage.
+	Querier() (Querier, error)
```

Comment:
No requirement to create something new, but implementations could need to create something new here. But yeah, fuzzy edge case I guess.